File: catalog/src/main/kotlin/com/victormonte/catalog/domain/Product.kt
package com.victormonte.catalog.domain

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = "products")
data class Product(
        @Id val _id: String,
        val id: String,
        val price_in_cents: Int,
        val title: String,
        val description: String,
        val discount: Discount)

File: discount/src/domain/model/Product.ts
import { Typegoose, prop } from 'typegoose';
import { Discount } from './discount';

export class Product extends Typegoose {

    @prop()
    id:String;

    @prop()
    priceInCents:number;

    @prop()
    title:String;

    @prop()
    description:String;

    @prop()
    discount:Discount;

}

export default new Product().getModelForClass(Product);